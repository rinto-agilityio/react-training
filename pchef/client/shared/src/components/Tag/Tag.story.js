// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Tag from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

storiesOf('Tag', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Tag
      onClose={action('clicked')}
    >
      Test
    </Tag>
  ))
  .add('Disabled', () => (
    <Tag
      onClose={action('clicked')}
      disabled="true"
    >
      Test
    </Tag>
  ))
