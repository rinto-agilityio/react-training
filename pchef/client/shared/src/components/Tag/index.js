// Libs
import React from 'react'
import { Chip } from 'react-native-paper';

type Props = {
  mode: string,
  customTextStyle?: {} | Array<{}>,
  customStyle?: {} | Array<{}>,
  children: string,
  onClose: () => void,
  disabled: boolean,
}

const Tag = ({
  children,
  mode,
  onClose,
  disabled,
}: Props) => (
  <Chip
    mode={mode}
    onClose={onClose}
    disabled={disabled}
  >
    { children }
  </Chip>
)

export default Tag
