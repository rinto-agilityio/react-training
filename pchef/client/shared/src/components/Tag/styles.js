// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

const styles = StyleSheet.create({
  tagStyles: {
    marginRight: METRICS.margin.sm,
    width: '100px',
  },
})

export default styles
