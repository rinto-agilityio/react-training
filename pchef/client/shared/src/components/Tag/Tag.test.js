// Components
import Tag from '.'

describe('Tag', () => {
  it('Renders correctly modal commponent', () => {
    const component = renderer.create(<Tag />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
