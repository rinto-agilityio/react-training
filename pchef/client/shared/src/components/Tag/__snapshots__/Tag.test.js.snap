// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Renders correctly modal commponent 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "backgroundColor": "rgba(235,235,235,1.00)",
      "borderBottomColor": "rgba(235,235,235,1.00)",
      "borderBottomLeftRadius": "16px",
      "borderBottomRightRadius": "16px",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "1px",
      "borderLeftColor": "rgba(235,235,235,1.00)",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "1px",
      "borderRightColor": "rgba(235,235,235,1.00)",
      "borderRightStyle": "solid",
      "borderRightWidth": "1px",
      "borderTopColor": "rgba(235,235,235,1.00)",
      "borderTopLeftRadius": "16px",
      "borderTopRightRadius": "16px",
      "borderTopStyle": "solid",
      "borderTopWidth": "1px",
      "marginRight": "5px",
      "width": "100px",
    }
  }
>
  <div
    aria-disabled={true}
    className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-userSelect-lrvibr"
    disabled={true}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      Object {
        "borderBottomLeftRadius": "16px",
        "borderBottomRightRadius": "16px",
        "borderTopLeftRadius": "16px",
        "borderTopRightRadius": "16px",
        "overflowX": "hidden",
        "overflowY": "hidden",
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-paddingHorizontal-s1qlax"
    >
      <div
        className="css-text-901oao css-textOneLine-bfa6kz r-lineHeight-135wba7 r-marginVertical-1p6iasa r-minHeight-1ceczpf r-textAlignVertical-1vzi8xi"
        dir="auto"
        style={
          Object {
            "color": "rgba(0,0,0,0.87)",
            "direction": "ltr",
            "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "marginLeft": "8px",
            "marginRight": "4px",
            "textAlign": "left",
          }
        }
      />
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-cursor-1loqt21 r-padding-tuq35u r-touchAction-1otgn73"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
          dir="auto"
          focusable={false}
          role="img"
          style={
            Object {
              "WebkitTransform": "scaleX(1)",
              "color": "rgba(0,0,0,0.54)",
              "fontFamily": "MaterialIcons",
              "fontSize": "16px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "transform": "scaleX(1)",
            }
          }
        >
          î—‰
        </div>
      </div>
    </div>
  </div>
</div>
`;
